src/main/kotlin/no/nav/familie/ef/sak/arbeidsfordeling/ArbeidsfordelingService.kt

```diff
@@ -1,31 +1,38 @@
 import no.nav.familie.ef.sak.opplysninger.personopplysninger.pdl.gjeldende
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.cache.CacheManager
 import org.springframework.stereotype.Component
 
 @Component
 class ArbeidsfordelingService(private val personService: PersonService,
-                              private val personopplysningerIntegrasjonerClient: PersonopplysningerIntegrasjonerClient) {
+                              private val personopplysningerIntegrasjonerClient: PersonopplysningerIntegrasjonerClient,
+                              @Qualifier("shortCache")
+                              private val cacheManager: CacheManager) {
 
     companion object {
 
         private const val MASKINELL_JOURNALFOERENDE_ENHET = "9999"
     }
 
     fun hentNavEnhet(ident: String): Arbeidsfordelingsenhet? {
-        val personMedRelasjoner = personService.hentPersonMedRelasjoner(ident)
-        val søkerIdentMedAdressebeskyttelse =
-                IdentMedAdressebeskyttelse(personMedRelasjoner.søkerIdent,
-                                           personMedRelasjoner.søker.adressebeskyttelse.gjeldende()?.gradering)
-        val identerMedAdressebeskyttelse = listOf(søkerIdentMedAdressebeskyttelse) +
-                                           personMedRelasjoner.barn.map {
-                                               IdentMedAdressebeskyttelse(it.key,
-                                                                          it.value.adressebeskyttelse.gjeldende()?.gradering)
-                                           }
-        val identMedStrengeste = finnPersonMedStrengesteAdressebeskyttelse(identerMedAdressebeskyttelse)
-        return personopplysningerIntegrasjonerClient.hentNavEnhet(identMedStrengeste ?: ident).firstOrNull()
+        return cacheManager.get("navEnhet", ident) {
+            val personMedRelasjoner = personService.hentPersonMedRelasjoner(ident)
+            val søkerIdentMedAdressebeskyttelse =
+                    IdentMedAdressebeskyttelse(personMedRelasjoner.søkerIdent,
+                                               personMedRelasjoner.søker.adressebeskyttelse.gjeldende()?.gradering)
+            val identerMedAdressebeskyttelse = listOf(søkerIdentMedAdressebeskyttelse) +
+                                               personMedRelasjoner.barn.map {
+                                                   IdentMedAdressebeskyttelse(it.key,
+                                                                              it.value.adressebeskyttelse.gjeldende()?.gradering)
+                                               }
+            val identMedStrengeste = finnPersonMedStrengesteAdressebeskyttelse(identerMedAdressebeskyttelse)
+            personopplysningerIntegrasjonerClient.hentNavEnhet(identMedStrengeste ?: ident).firstOrNull()
+        }
     }
 
     fun hentNavEnhetIdEllerBrukMaskinellEnhetHvisNull(personIdent: String): String {
```

Comment:
Alternativet i denne hadde kunnet vært å ha en extension function på Arbeidsfordelingsenhet for `hentNavEnhetIdEllerBrukMaskinellEnhetHvisNull`, og bruke @Cacheable på `hentNavEnhet`